apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/Emulator/metadata_server.py

```diff
@@ -0,0 +1,353 @@
+    
+    if isinstance(result, IndexedList):
+      try:
+        index = int(component)
+      except ValueError:
+        return not_found_error
+      
+      if 0 <= index and index < len(result):
+        result = result[index]
+      else:
+        return not_found_error
+    else:
+      component = component.replace(unicode(numeric_project_id), numeric_project_id_token)
+      component = component.replace(project_id, project_id_token)
+      if not component in result:
+        return not_found_error
+
+      result = result[component]
+
+  if result == numeric_project_id_token:
+    result = numeric_project_id
+
+  if is_token_request:
+    # Even though this acts like a directory, trailing slashes are not allowed for tokens.
+    if ends_with_slash:
+      return not_found_error
+
+    is_directory = True
+    is_json = (alt != 'text')
+    recursive = True
+  else:
+    # recursive directory requests and instance/tags default to json
+    is_directory = isinstance(result, OrderedDict) or isinstance(result, IndexedList) or isinstance(result, list)
+    is_json = (alt == 'json' or ((component == 'tags' or (recursive and is_directory)) and alt != 'text'))
+    if ends_with_slash and not is_directory:
+      return not_found_error
+
+  if wait_for_change:
+    # When waiting for changes on non-list directories, it must be a recursive request
+    if is_directory and not isinstance(result, list) and not recursive:
+      return bad_request_error
+
+    if timeout_sec < 0:
+      return formatResponse('Error: indefinitely waiting for changes is not supported',
+                            400, 
+                            common.content_type_html)
+    time.sleep(timeout_sec)
+
+  if recursive:
+    if is_json:
+      result = json.dumps(result, cls=MetadataJSONEncoder, separators=(',', ':'))
+    else:
+      result = convertToTextRecursive('', result)
+  else:
+    if is_directory and alt and not isinstance(result, list):
+      return bad_request_error
+
+    if isinstance(result, OrderedDict):
+      result = formatOrderedDictContents(result)
+    elif isinstance(result, IndexedList):
+      result = result.formatContents()
+    elif isinstance(result, list):
+      if is_json:
+        result = json.dumps(result, separators=(',', ':'))
+      else:
+        result = ''.join([s + '\n' for s in result])
+    elif isinstance(result, unicode):
+      if is_json:
+        result = '"' + result + '"'
+    else:
+      result = unicode(result)
+
+  return formatResponse(result.replace(numeric_project_id_token,
+                                       unicode(numeric_project_id)).replace(project_id_token,
+                                                                            project_id),
+                        200,
+                        common.content_type_json if is_json else common.content_type_text)
+
+if __name__ == '__main__':
+  test_project_id = project_id
+  try:
+    project_id = os.environ['TEST_PROJECT']
```

Comment:
if you're just going to pass on the KeyError, you can just do `project_id = os.environ.get('TEST_PROJECT')`